### PR TITLE
Improve README.md (Virtualenv paragraph, minor fixes)

### PR DESCRIPTION
**Pull request by ronjouch**
* #71 - Add a 'Virtualenv' paragraph mentioning a good fool-proof setup
* Use proper Atom terminology:
  * Configuration panel -> Settings panel
  * Plugin -> Package
* English typos
* Markdown typos

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,34 @@
 linter-flake8
 =============
 
-linter-flake8 is a [flake8](https://pypi.python.org/pypi/flake8 provider for [Linter](https://github.com/AtomLinter/Linter).
+linter-flake8 is a [flake8](https://pypi.python.org/pypi/flake8) provider for [Atom-Linter](https://github.com/AtomLinter/Linter).
```
Comment (Arcanemagus) [inline README.md]: The package is just called `Linter`, although the eventual goal is to have it integrated into Atom :wink:.

File: README.md [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
 
 ![img](https://cloud.githubusercontent.com/assets/4278113/8768482/52f975c6-2e3f-11e5-87e4-27c8359fd36c.gif)
 
 ### Requirements
-Linter package must be installed in order to use this plugin. If it's not
-installed, please follow the instructions [here](https://github.com/AtomLinter/Linter).
+The `Linter` package must be installed in order to use this package. If it's not installed, please follow the [Linter instructions](https://github.com/AtomLinter/Linter).
```
Comment (Arcanemagus) [inline README.md]: \[installation instructions\] instead? They aren't the instructions on how to lint.

Also you might as well link directly to the installation instructions: https://github.com/atom-community/linter#how-to--installation

File: README.md [hunk 3]
```diff
@@ -1,30 +1,34 @@
...
-Install [flake8](https://pypi.python.org/pypi/flake8) by typing the following
-in a terminal:
+First, ensure that [flake8](https://pypi.python.org/pypi/flake8) is installed on your system. If not, install it by typing the following in a terminal:
    ```
    pip install flake8
    ```
 
-Install plugin by typing:
+Install package by typing:
    ```
```
Comment (Arcanemagus) [inline README.md]: I usually put `ShellSession` here ([source](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L3172)), but it isn't important.

Comment (ronjouch) [inline README.md]: For one-liners, adds too much noise when reading the plain .md file IMHO.

File: README.md (hunk 2 above)
Comment (ronjouch) [inline README.md]: Yup, just wanted a better link name than "here". Done.

File: README.md (hunk 1 above)
Comment (ronjouch) [inline README.md]: There are lots of `Linter`s, one might confuse this e.g. with Sublime Linter; just wanted to mention we're talking Atom here. Will reformulate to "linter-flake8 is a [flake8](https://pypi.python.org/pypi/flake8) provider for Atom [Linter](https://github.com/atom-community/linter) package"

File: README.md [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
-
-Install [flake8](https://pypi.python.org/pypi/flake8) by typing the following
-in a terminal:
+First, ensure that [flake8](https://pypi.python.org/pypi/flake8) is installed on your system. If not, install it by typing the following in a terminal:
    ```
    pip install flake8
    ```
 
-Install plugin by typing:
+Install package by typing:
```
Comment (Arcanemagus) [inline README.md]: "Then install this package by typing"?
The current wording doesn't flow right to me at least.

File: README.md [hunk 5]
```diff
@@ -1,30 +1,34 @@
...
    ```
-   $ apm install linter-flake8
+   apm install linter-flake8
    ```
 
-##### Bult-in docstrings check (Optional)
+#### Virtualenv
+
+To ensure `flake8` targets the proper Python version for each project you work on, it is highly recommended to install flake8 in each project [virtualenv](https://virtualenv.pypa.io/en/latest/):
```
Comment (Arcanemagus) [inline README.md]: project's

File: README.md [hunk 6]
```diff
@@ -1,30 +1,34 @@
...
+
+* If you are okay with installing flake8 in each virtualenv, just do that.
+* If you'd rather use system-wide flake8 versions (e.g. `flake8` and `flake8-python2`), create a `flake8` symlink in each virtualenv pointing to your desired flake8 version. For example: `ln -s /usr/bin/flake8-python2 ~/.virtualenvs/random_python2_project/bin/flake8`
+
+Then [activate your virtualenv](https://virtualenv.pypa.io/en/latest/userguide.html#activate-script) from the command line before starting Atom, and you're good to go!
+
+##### Built-in docstrings check (Optional)
 To include built-in docstrings (pep257) support you will also need to install:
    ```
    pip install flake8-docstrings
```
Comment (Arcanemagus) [inline README.md]: We might also mention [`flake8-pep257`](https://pypi.python.org/pypi/flake8-pep257) which does the same thing.

`flake8-docstrings` actually doesn't work with this package right now (at least on Windows), although the author is [working on it](https://gitlab.com/pycqa/flake8-docstrings/issues/2).

File: README.md [hunk 7]
```diff
@@ -41,14 +45,14 @@ To support [OpenStack Style Guidelines](http://google-styleguide.googlecode.com/
   ```
 
 ### Settings
-You can configure linter-flake8 as nearly all [Atom](https://atom.io/) modules by editing plugin settings in *Atom->Preferences->Packages->linter-flake-8*.
+You can configure linter-flake8 as nearly all [Atom](https://atom.io/) modules by editing the package Settings in *Atom -> Preferences -> Packages -> linter-flake8*.
 
 ![img](https://cloud.githubusercontent.com/assets/4278113/8768488/91fffae2-2e3f-11e5-9783-9513b882cbc0.png)
 
-Or if you prefere you can use Atom *config.cson* file *~/.atom/config.cson* (click *Open Your Config*
+Or if you prefer you can use Atom *config.cson* file *~/.atom/config.cson* (click *Open Your Config*
```
Comment (Arcanemagus) [inline README.md]: Atom's

File: README.md [hunk 8]
```diff
@@ -41,14 +45,14 @@ To support [OpenStack Style Guidelines](http://google-styleguide.googlecode.com/
...
 in *Atom* menu).
 
-In configuration you can specify executable directory if node hasn't it in **$PATH**. Example:
+In the Settings panel you can specify executable directory if node hasn't it in **$PATH**. Example:
```
Comment (Arcanemagus) [inline README.md]: "...if Atom isn't able to determine it automatically. Example:" instead?

(At the very least "hasn't it in" needs to be re-written)

File: README.md [hunk 9]
```diff
@@ -56,6 +60,6 @@ In configuration you can specify executable directory if node hasn't it in **$PA
 ```
 
 ### Project configuration files
-Linter also supports reading flake8 configuration files. To use them, you need specify their names manually into *Config Files Names* in plugin settings. Please take a note, that  plugin settings from configuration panel will be **ignored** if configuration file was found.
+Linter also supports reading flake8 configuration files. To use them, you need specify their names manually into *Config Files Names* in package settings. Please note that package settings from the Settings panel will be **ignored** if a configuration file was found.
```
Comment (Arcanemagus) [inline README.md]: flake8 -> \`flake8\`

**Comment by Arcanemagus** [comment]
That's all that I saw, thanks for doing this!

File: README.md (hunk 7 above)
Comment (Arcanemagus) [inline README.md]: Or: "use the Atom *config.cson* file"

**Comment by ronjouch** [comment]
All fixed; amended my commit and re-pushed.

**Comment by Arcanemagus** [comment]
Mind updating the OpenStack Style guide link while you are at it? It's currently pointing to googlecode.com and should be poinint to this location:
http://google.github.io/styleguide/pyguide.html

**Comment by ronjouch** [comment]
> Mind updating the OpenStack Style guide link while you are at it? It's currently pointing to googlecode.com and should be poinint to this location:
> http://google.github.io/styleguide/pyguide.html

Sure, done, thanks for the new link.

**Comment by badray** [comment]
Hi! Thanks for your work! But unfortunately I can't do anything with this merge(Review, eventualy merge and release) until sunday 30.08. Having crunch time in my work till then :/

**Comment by ronjouch** [comment]
No problem, no one is paying you for maintaining this open-source project; review it when you have the time and good luck with your job!

**Comment by Arcanemagus** [comment]
Just realized that this and #79 conflict on the readme changes. @badray can you review this and I'll update #79 after this gets merged, or just resubmit it.

**Comment by Arcanemagus** [comment]
@ronjouch can you rebase this on the current master?

**Comment by ronjouch** [comment]
> @ronjouch can you rebase this on the current master?

@Arcanemagus nope, trying to rebase yields the whole file in conflict; not going to re-do this all over again.

**Comment by Arcanemagus** [comment]
Okay, I'll cherry pick anything that still needs updating from these changes at some point, thanks!

**Pull request closed without merging**